adapters/richaudience/richaudience.go

```diff
@@ -0,0 +1,299 @@
+package richaudience
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/metrics"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type richaudienceRequest struct {
+	ID     string             `json:"id,omitempty"`
+	Imp    []openrtb2.Imp     `json:"imp,omitempty"`
+	User   richaudienceUser   `json:"user,omitempty"`
+	Device richaudienceDevice `json:"device,omitempty"`
+	Site   richaudienceSite   `json:"site,omitempty"`
+	Test   int8               `json:"test,omitempty"`
+}
+
+type richaudienceUser struct {
+	BuyerUID string              `json:"buyeruid,omitempty"`
+	Ext      richaudienceUserExt `json:"ext,omitempty"`
+}
+
+type richaudienceUserExt struct {
+	Eids    []openrtb_ext.ExtUserEid `json:"eids,omitempty"`
+	Consent string                   `json:"consent,omitempty"`
+}
+
+type richaudienceDevice struct {
+	IP   string `json:"ip,omitempty"`
+	IPv6 string `json:"ipv6,omitempty"`
+	Lmt  int8   `json:"lmt,omitempty"`
+	DNT  int8   `json:"dnt,omitempty"`
+	UA   string `json:"ua,omitempty"`
+}
+type richaudienceSite struct {
+	Domain string `json:"domain,omitempty"`
+	Page   string `json:"page,omitempty"`
+}
+
+// Builder builds a new instance of the RichAudience adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	richaudienceRequest := richaudienceRequest{}
+
+	raiHeaders := http.Header{}
+
+	setHeaders(&raiHeaders)
```

Comment:
I like the names and organization of the helper methods. Consider changing `setHeaders` to something like `buildHeaders` and embed in the creation of the HTTP request:

```
requestData := &adapters.RequestData{
  Method:  "POST",
  Uri:     a.endpoint,
  Body:    req,
  Headers: buildHeaders(),
}
```

Reply:
Thank you for your comments, but I am not going to change this.